doc/string.rdoc

```diff
@@ -47,4 +47,81 @@ class String
   def initialize(str = '', encoding: nil, capacity: nil)
     Primitive.rb_str_init(str, encoding, capacity)
   end
+
+  #  call-seq:
+  #    split(pattern = $FIELD_SEPARATOR, limit = nil) -> array
+  #    split(pattern = $FIELD_SEPARATOR, limit = nil) {|sub| block } -> object
+  #
+  #  Returns an array of substrings of +self+;
+  #  the substrings are copied from +self+, based on the given arguments.
+  #
+  #  When argument +pattern+ is $FIELD_SEPARATOR:
+  #
+  #  - If $FIELD_SEPARATOR is +nil+ (its default value),
+  #    the split occurs just as if +pattern+ were given as a space character
+  #    (see below).
+  #
+  #  - If $FIELD_SEPARATOR is a string,
+  #    the split ocurs just as if +pattern+ were given as that string
+  #    (see below).
+  #
+  #  When argument +pattern+ is <tt>' '</tt> and argument +limit+ is +nil+,
+  #  the split occurs at each sequence of whitespace:
+  #
+  #    'abc def ghi'.split(' ')          # => ["abc", "def", "ghi"]
+  #    "abc \n\tdef\t\n  ghi".split(' ') # => ["abc", "def", "ghi"]
+  #    'abc  def   ghi'.split(' ')       # => ["abc", "def", "ghi"]
+  #    ''.split(' ')                     # => []
+  #
+  #  When argument +pattern+ is a string different from <tt>' '</tt>
+  #  and argument +limit+ is +nil+,
```

Comment:
I'm not sure if we should have examples that treat a single space as a special case, it works the same just like any other character.

Instead, I think we should mention that `$;` is `nil` by default. And if `pattern` is `nil`, then it's treated as if `pattern` is `' '`.

Reply:
"a\nb".split(' ')# => ["a", "b"] splits on newline.

Reply:
Ah sorry I didn't read clearly enough. I didn't realize splitting with a `pattern` of `' '` splits on all types of whitespace.